lib/src/main/java/graphql/nadel/validation/NadelHydrationWhenConditionValidation.kt

```diff
@@ -0,0 +1,71 @@
+package graphql.nadel.validation
+
+import graphql.Scalars
+import graphql.nadel.dsl.UnderlyingServiceHydration
+import graphql.schema.GraphQLFieldDefinition
+import graphql.schema.GraphQLFieldsContainer
+import graphql.schema.GraphQLScalarType
+import java.math.BigInteger
+
+internal class NadelHydrationWhenConditionValidation() {
+
+    fun validateHydrationWhenConditionInput(
+        parent: NadelServiceSchemaElement,
+        overallField: GraphQLFieldDefinition,
+        hydration: UnderlyingServiceHydration,
+    ): NadelSchemaValidationError? {
+        if (hydration.conditionalHydration == null) {
+            return null
+        }
+
+        val whenConditionSourceFieldName: String = hydration.conditionalHydration?.get("sourceField") as String
+        val whenConditionSourceField: GraphQLFieldDefinition = (parent.overall as GraphQLFieldsContainer).getField(whenConditionSourceFieldName)
+
+        if (whenConditionSourceField.type !is GraphQLScalarType){
+            return return NadelSchemaValidationError.WhenConditionUnsupportedFieldType(whenConditionSourceFieldName, "", overallField)
```

Comment:
Did you leave the type name here blank on purpose?

You should be able to get a type name using something like `GraphQLTypeUtil.simplePrint(whenConditionSourceField.type)`